src/main/scala/cognite/spark/v1/SequenceRowsRelation.scala

```diff
@@ -164,9 +166,10 @@ class SequenceRowsRelation(config: RelationConfig, sequenceId: CogniteId)(val sq
         case _ => throw SparkSchemaHelperRuntime.badRowError(row, "rowNumber", "Long", "")
       }
 
-      val externalId = row.get(externalIdIndex) match {
-        case x: String => x
-        case _ => throw SparkSchemaHelperRuntime.badRowError(row, "externalId", "String", "")
+      val externalId = sequenceInfo.externalId match {
```

Comment:
This is now taking the externalId from `sequenceInfo` instead of the current Spark Row. The `sequenceInfo` is global for all rows and we should basically only use it for read and schema inferrence